Sources/Rendering/Core/HardwareSelector/example/index.js

```diff
@@ -151,13 +150,12 @@ renderWindow.render();
 // Create hardware selector
 // ----------------------------------------------------------------------------
 
-const hardwareSelector = vtkOpenGLHardwareSelector.newInstance({
-  captureZValues: true,
-});
+const hardwareSelector = apiSpecificRenderWindow.getSelector();
+hardwareSelector.setCaptureZValues(true);
 hardwareSelector.setFieldAssociation(
   FieldAssociations.FIELD_ASSOCIATION_POINTS
 );
-hardwareSelector.attach(openGLRenderWindow, renderer);
+hardwareSelector.attach(apiSpecificRenderWindow, renderer);
```

Comment:
Should the `hardwareSelector` already have a reference to `apiSpecificRenderWindow` since it was created from it? Could we just attach/setRenderer?

Reply:
Yes we could. That API is old so I just left it as it wast for backwards compatibility and the off chance someone explicitly creates a hardware selector bypassing window.getHardwareSelector

Reply:
Should we have `setView`, `setRenderer` and `attach(view, renderer)` (as-is) while deprecating the `attach` one?

Reply:
Yes, that makes more sense. Or we could have renderer be an argument to getSourceDataAsync as what it does is specify what renderer to do the selection on. Which do you think is better?

Reply:
I think it will be less error prone to pass the renderer at the time you call `getSourceDataAsync`. That way the user never needs to remember to bind/attach/set anything...

Reply:
I'm not a big fan of async in the method name... I don't know if we need to be explicit in the name that we return a promise...

Reply:
Well we should have a convention but I'm fine either way. I just followed WebGPU's approach which appends Async onto any async methods. So feel free to make a call and let me know and I'll update to match.

Reply:
If WebGPU does that, I'm fine keeping the same structure.